src/core/qgspointlocator.cpp

```diff
@@ -636,6 +636,11 @@ QgsPointLocator::QgsPointLocator( QgsVectorLayer *layer, const QgsCoordinateRefe
   connect( mLayer, &QgsVectorLayer::featureDeleted, this, &QgsPointLocator::onFeatureDeleted );
   connect( mLayer, &QgsVectorLayer::geometryChanged, this, &QgsPointLocator::onGeometryChanged );
   connect( mLayer, &QgsVectorLayer::dataChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::rendererChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::styleChanged, this, &QgsPointLocator::destroyIndex );
+  connect( mLayer, &QgsVectorLayer::layerModified, this, &QgsPointLocator::destroyIndex );
+  // used when a style is (de)activated
+  connect( mLayer, &QgsVectorLayer::repaintRequested, this, &QgsPointLocator::destroyIndex );
```

Comment:
Not sure about destroying the index on a repaint and layer modified - these signals are thrown whenever a feature is edited in a layer - so by destroying the index when that occurs it's forcing a full rebuild, instead of the faster update handling done by onFeatureDeleted/onGeometryChanged. (I'd also suggest working on a test data set with 100k+ features while test running this.. that'll be bound to reveal any performance regressions!)

I think you'll need to fix the missing styleChanged signal instead.